compiler/src/dmd/iasmgcc.d

```diff
@@ -143,6 +151,30 @@ public void gccAsmSemantic(CAsmDeclaration ad, Scope* sc)
 
 private:
 
+
+/***********************************
+ * Issue error if the current token is not `value`.
+ * Otherwise, advance to next token.
+ * Params:
+ *      p = parser state
+ *      value = token value to compare with
+ * Returns:
+ *      true    advanced to next token
+ *      false   error was issued
+ */
+bool requireToken(Parser)(Parser p, TOK value)
```

Comment:
Yes, this is almost the same as `void check(TOK)` - key differences, apart from bool return, the token is only advanced on success.